src/hotspot/share/opto/mulnode.cpp

```diff
@@ -1296,59 +1308,160 @@ const Type* LShiftLNode::Value(PhaseGVN* phase) const {
   return TypeLong::make( (jlong)r1->get_con() << (jint)shift );
 }
 
-//=============================================================================
-//------------------------------Identity---------------------------------------
-Node* RShiftINode::Identity(PhaseGVN* phase) {
+Node *RShiftNode::IdealIL(PhaseGVN* phase, bool can_reshape, BasicType bt) {
+  // Inputs may be TOP if they are dead.
+  const TypeInteger* t1 = phase->type(in(1))->isa_integer(bt);
+  if (t1 == nullptr) {
+    return NodeSentinel;        // Left input is an integer
+  }
+  const TypeInteger* t3;  // type of in(1).in(2)
```

Comment:
I know that you only moved this code, but it looks bad :see_no_evil: 
For one, why is it defined up here already when it is only used 10 lines later?
And why not give it a better name so we don't need the comment?
```suggestion
```